src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -1024,6 +1024,16 @@ internal static object GetModelStateValue(ViewContext viewContext, string key, T
                         attributeValue = useViewData ? EvalString(viewContext, fullName, format) : valueParameter;
                     }
 
+                    object typeAttributeValue = null;
```

Comment:
this added block is going to confuse some future developer since multiple breaks from the same `case` is unusual.  suggest reversing the tests (in some way) or adding a `var addValue = true;` to control merging the `value` attribute.

also need a comment explaining why not adding the `value` attribute is correct in these cases.